graphqlapi/dynamic_generation_meta_fetch.go

```diff
@@ -0,0 +1,650 @@
+			},
+		},
+		"occurs": &graphql.Field{
+			Name:        mergeStrings("Meta", class.Class, property.Name, "TopOccurencesOccurs"),
+			Description: "number of occurrance",
+			Type:        graphql.String,
+			Resolve: func(p graphql.ResolveParams) (interface{}, error) {
+				return nil, fmt.Errorf("Not supported")
+			},
+		},
+	}
+	metaFetchMetaPointing := graphql.ObjectConfig{
+		Name:        mergeStrings("Meta", class.Class, property.Name, "TopOccurencesObj"),
+		Fields:      metaFetchMetaPointingFields,
+		Description: "most frequent property values",
+	}
+	return graphql.NewObject(metaFetchMetaPointing), nil
+}
+
+func genMetaClassIntPropertyFields(class *models.SemanticSchemaClass, property *models.SemanticSchemaClassProperty) (*graphql.Object, error) {
+
+	metaFetchMetaIntFields := graphql.Fields{
+		"sum": &graphql.Field{
+			Name:        mergeStrings("Meta", class.Class, property.Name, "Sum"),
+			Description: "sum of values of found instances",
+			Type:        graphql.Float,
+			Resolve: func(p graphql.ResolveParams) (interface{}, error) {
+				return nil, fmt.Errorf("Not supported")
+			},
+		},
+		"type": &graphql.Field{
+			Name:        mergeStrings("Meta", class.Class, "Type"),
+			Description: "datatype of the property",
+			Type:        graphql.String,
+			Resolve: func(p graphql.ResolveParams) (interface{}, error) {
+				return nil, fmt.Errorf("Not supported")
+			},
+		},
+		"lowest": &graphql.Field{
+			Name:        mergeStrings("Meta", class.Class, "Lowest"),
+			Description: "Lowest value occurence",
+			Type:        graphql.Float,
+			Resolve: func(p graphql.ResolveParams) (interface{}, error) {
+				return nil, fmt.Errorf("Not supported")
+			},
+		},
+		"highest": &graphql.Field{
+			Name:        mergeStrings("Meta", class.Class, "Highest"),
+			Description: "Highest value occurence",
+			Type:        graphql.Float,
+			Resolve: func(p graphql.ResolveParams) (interface{}, error) {
+				return nil, fmt.Errorf("Not supported")
+			},
+		},
+		"average": &graphql.Field{
+			Name:        mergeStrings("Meta", class.Class, "Average"),
+			Description: "average number",
+			Type:        graphql.Float,
+			Resolve: func(p graphql.ResolveParams) (interface{}, error) {
+				return nil, fmt.Errorf("Not supported")
+			},
+		},
+		"counter": &graphql.Field{
+			Name:        mergeStrings("Meta", class.Class, "Counter"),
+			Description: "total amount of found instances",
+			Type:        graphql.Int,
+			Resolve: func(p graphql.ResolveParams) (interface{}, error) {
+				return nil, fmt.Errorf("Not supported")
+			},
+		},
+	}
+	metaFetchIntProperty := graphql.ObjectConfig{
+		Name:        mergeStrings("Meta", class.Class, property.Name, "Obj"),
+		Fields:      metaFetchMetaIntFields,
+		Description: "Property meta information",
+	}
+	return graphql.NewObject(metaFetchIntProperty), nil
+}
+
+// a duplicate of the int function, this is a separate function to account for future expansions of functionality
```

Comment:
After moving from float to int in the function just before this one, this comment will be obsolete.